File: csa-official-backend/src/main/java/com/csa/official/modules/sys/entity/Carousel.java
package com.csa.official.modules.sys.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("sys_carousel")
public class Carousel implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String imgUrl;
    private String targetUrl;
    private String title;
    private Integer sortOrder;
    private Integer status; // 1:启用, 0:禁用

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    @TableLogic
    private Integer deleted;
}

File: csa-official-frontend/src/services/carousel.ts
import api from "@/lib/axios"

export interface CarouselAdminItem {
  id: number
  imgUrl: string
  targetUrl: string | null
  title: string
  sortOrder: number
  status: number
  createTime: string | null
  updateTime: string | null
}

export interface SaveCarouselPayload {
  id?: number
  imgUrl: string
  targetUrl?: string | null
  title: string
  sortOrder: number
  status: number
}

export const carouselService = {
  list: () =>
    api.get<CarouselAdminItem[], CarouselAdminItem[]>("/api/sys/carousel/list"),

  save: (data: SaveCarouselPayload) =>
    api.post<string, string>("/api/sys/carousel/save", data),

  remove: (id: number) =>
    api.post<string, string>("/api/sys/carousel/delete", null, {
      params: { id },
    }),
}
